### DISCUSSION TITLE
Add lock to Button-Screen

### DISCUSSION THREAD

**Discussion by musterzwerg** [Blackymas/NSPanel_HA_Blueprint#1178] [Ideas]
Thank you for your amazing work. Everthing works so far. I have a Nuki lock in my home and I would love to have the possibility to use the lock also on the panel.

**Comment by edwardtfn** [comment]
I don't have any lock, so I will need your help here... How is the user interface in Home Assistant when you are playing with the lock?

**Comment by musterzwerg** [comment]
Hi, actually its similar to a switch.
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/63115735/95f8fcf5-c961-4898-97c7-41d0a65837d2)

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/63115735/900c2343-8044-40b3-af9f-7bf4b71ad6d7)

https://developers.home-assistant.io/docs/core/entity/lock/

Thank you

**Reply by bertybassett** [reply]
Scripts.yaml
unlock_frontdoor:
  mode: restart
  sequence:
  - service: lock.unlock
    target:
      entity_id: lock.nuki
      
lock_frontdoor:
  mode: restart
  sequence:
  - service: lock.lock
    target:
      entity_id: lock.nuki  

then added these as custom buttons

Looks rubbish and not what I wanted but does kind of work

**Reply by edwardtfn** [reply]
This is something pretty easy to add to the panel, but I still have some questions here:

1. On the button page, the button have an state "on" (white background) or "off" (gray background). Which of the states of a lock should correspond to "on" and which to "off"?
2. If you go to Developer Tools and then select the tab "State", what are the different states for a "lock" entity?
3. Could you please share a screenshot of that page showing attributes (regardless the state for the screenshot).

**Reply by bertybassett** [reply]
1) don’t think it matters, well doesn’t matter to me. But if you forced me to say I’d say white for unlocked and grey for locked. In ha it’s green for locked and red for unlocked. 

2) locked and unlocked

3)
![IMG_0876](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/8549d6de-55f9-4cd3-98a0-4c044ab84213)

![IMG_0877](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/882a528e-9bd0-4e3f-a0c3-c5b2ff18f047)

![IMG_0878](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/13233552-1eff-49d8-97ad-7c99f8598d88)

if you need any more info just shout.

**Reply by edwardtfn** [reply]
Thanks a lot for all that info!!
I've implemented this in `dev`, however I cannot test it.
If someone can test it and everything goes well this should be part of v4.3 to be released very soon.
I will close this for now, to make easier managing the backlog, but please feel free to add comments here or reopen-it if it isn't solved.

**Reply by bertybassett** [reply]
How do I change to dev? reflash it and change main to dev in the esphome  page?

**Reply by edwardtfn** [reply]
This is probably needing some update, but it might help: [Guide to Using Different Versions of This Project ](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/different_version.md)

**Reply by bertybassett** [reply]
I’ll give it a try tonight.

**Reply by plasterer41** [reply]
Edward, can you private message your PayPal address also as I would like to make a donation to buy you a beer 🍺 for all the help you have given me.

**Comment by edwardtfn** [comment]
- #375
- #606 
- #1723

**Comment by bertybassett** [comment]
updated to 4.3dev2 and the lock entity called lock.nuki works as both a chip and a button.

However in HA the default lock icon changes as you press it, would it be possible have these 2 lock icons, where 

unlocked = 

<img width="117" alt="Screenshot 2024-03-06 at 16 18 28" src="https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/776a305c-b0d2-45f0-9622-36b19a686a2d">

and locked =  
<img width="128" alt="Screenshot 2024-03-06 at 16 35 53" src="https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/bd2401eb-9a1e-4534-8148-b5dcf0fc33ff">



This is only needed on the button page (not chip as that only displays one state).

**Reply by bertybassett** [reply]
Also Home Assistant automatically changes the colour of the icons, it  green for locked and red for unlocked if the blueprint could reflect this that would be amazing.


What is your PayPal and I will send you some cash for a few coffee, coke or beer tokens

**Reply by bertybassett** [reply]
Also I know its a dev but the thermostat card is totally broken.

**Reply by edwardtfn** [reply]
- #1862 

Both the climate thing and also the state based icon/color for lock have been fixed, but you should wait for HA v2024.3 to be released in a couple of hours as other change will require that.

**Reply by bertybassett** [reply]
you legend!!!!!!!!!!!

**Comment by bertybassett** [comment]
update HA and Nspanel TFT, ESPHome, and blueprint 

this works however the icons are the wrong way round compared to HA.

Let me grab a screen shot to show you what is wrong

**Reply by bertybassett** [reply]
![IMG_0887](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/11979c7c-4526-4179-9067-04f0fe04e796)
![IMG_0888](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/f707c954-1141-4e65-a48d-0970061babdd)

**Reply by bertybassett** [reply]
would it be possible to use this as the unlocked icon as the tiny screen make its hard to see the default icon https://pictogrammers.com/library/mdi/icon/lock-open-variant/
<img width="117" alt="Screenshot 2024-03-07 at 10 36 21" src="https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/d132f59c-76e6-47b3-944b-fad110a0e506">


Lastly I am not sure we need the background colour change when switched on because a lock does not really have an on state unlike a bulb. I think just changing the icons colour from green to red would is enough.

**Reply by bertybassett** [reply]
Btw climate is working now.

**Reply by edwardtfn** [reply]
Icon fixed in `dev`. 😉

**Reply by bertybassett** [reply]
I'll go test.

**Reply by bertybassett** [reply]
what version number should it be??? I get 4.3.dev3 and its still wrong


![IMG_0901](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/bf3e0928-6e1f-436d-b9da-9829a01ac762)
![IMG_0902](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/5606175/77995795-90df-4184-b838-f77256439831)

**Reply by edwardtfn** [reply]
Could you please try with v4.3dev4?

**Reply by edwardtfn** [reply]
Please make sure you update the blueprint also.

**Reply by bertybassett** [reply]
had to do a clean build

**Reply by bertybassett** [reply]
All working now on.

TFT: 4.3dev3
ESPhome: 4.3dev4
Blueprint: 4.3.dev4
